vowpalwabbit/v_array.h

```diff
@@ -377,12 +378,15 @@ struct v_array<T, typename std::enable_if<std::is_trivially_copyable<T>::value>:
     else  // size = 1, ele = 1, _begin[0] = 0
       return b;
   }
+
+  VW_DEPRECATED("Use insert and std::upper_bound instead. This will be removed in VW 9.0.")
```

Comment:
to reproduce this behavior, you'd actually have to do a lower_bound and check if the value is equal.. we might need to keep this function around because there's no easy way to reproduce exactly what it does...

Reply:
Hmm, you can do an upper_bound then check prev if it is equal to the value to be inserted.

Reply:
lower_bound gives you the first index thats >= the target, which will either give you an iterator to the value you want or the upper_bound if it doesn't exist

Reply:
If you use upper bound and go back one then you only need to traverse the list once compared to if you use lower_bound to determine membership then upper bound to find the iterator to insert at